opsdroid/connector/matrix/connector.py

```diff
@@ -216,9 +218,14 @@ async def connect(self):
             self.connection.user_id = self.mxid
 
         elif self.mxid is not None and self.password is not None:
-            login_response = await self.connection.login(
-                password=self.password, device_name=self.device_name
-            )
+            try:
+                login_response = await self.connection.login(
+                    password=self.password, device_name=self.device_name
+                )
+            except Exception:
+                _LOGGER.error("connexion error")
```

Comment:
```suggestion
                _LOGGER.error("Logging into matrix homeserver failed.")
```

Reply:
I don't think the error case is handled properly here. If a request nio makes fails, it does not raise an excepiton it returns an [error class](https://github.com/poljar/matrix-nio/blob/main/nio/client/async_client.py#L960).

This also points out that there should be a test for this.